contribs/gmf/src/query/gridComponent.js

```diff
@@ -675,13 +686,18 @@ exports.Controller_.prototype.makeGrid_ = function(data, source) {
  * @return {?ngeo.grid.Config} Grid config.
  * @private
  */
-exports.Controller_.prototype.getGridConfiguration_ = function(
-  data) {
+exports.Controller_.prototype.getGridConfiguration_ = function(data) {
   googAsserts.assert(data.length > 0);
   const clone = {};
   Object.assign(clone, data[0]);
   delete clone.ol_uid;
-  const columns = Object.keys(clone);
+
+  const dataSource = this.ngeoDataSources_.collection.getArray().find(
+    ds => ds.id == clone.ngeo_datasource_id);
+  delete clone.ngeo_datasource_id;
+
+  const columns = dataSource && dataSource.columnsOrder.filter(key => key in clone)
```

Comment:
if `dataSource.columnsOrder` is undefined (i.e. attributesColumn) then this would fail.  Let's add `&& dataSource.columnsOrder`.

Also, personnal opinion: I like one liner if/else when they are short. This one is pretty long, so I wouldn't mind a regular if/else statement instead to make the code easier to read.